e2e/callback_test.go

```diff
@@ -195,351 +196,322 @@ func (suite *CbTestSuite) SetupTest() {
-	_, err := suite.chainC.SendMsgs(&wasmtypes.MsgExecuteContract{
-		Sender:   suite.chainC.SenderAccount.GetAddress().String(),
-		Contract: suite.testerC.String(),
-		Msg:      []byte(msg),
-		Funds:    []sdk.Coin{},
-	})
-	require.NoError(suite.T(), err)
-
-	if relay {
-		suite.coordinator.UpdateTime()
-		suite.coordinator.RelayAndAckPendingPackets(suite.pathAC.Invert())
-	}
-}
-
-func queryTesterSent(t *testing.T, chain *wasmibctesting.TestChain, tester string) string {
-	resp := TesterResponse{}
-	testerSentQuery := TesterSentQuery{
-		GetSentCallback: EmptyData{},
-	}
-	err := chain.SmartQuery(tester, testerSentQuery, &resp)
-	require.NoError(t, err)
-	return *resp.Owner
-}
-
-func queryTesterReceived(t *testing.T, chain *wasmibctesting.TestChain, tester string) string {
-	resp := TesterResponse{}
-	testerReceivedQuery := TesterReceivedQuery{
-		GetReceivedCallback: EmptyData{},
-	}
-	err := chain.SmartQuery(tester, testerReceivedQuery, &resp)
-	require.NoError(t, err)
-	return *resp.Owner
-}
-
-func queryTesterNftContract(t *testing.T, chain *wasmibctesting.TestChain, tester string) string {
-	resp := ""
-	testerReceivedQuery := TesterNftContractQuery{
-		GetNftContract: EmptyData{},
-	}
-	err := chain.SmartQuery(tester, testerReceivedQuery, &resp)
-	require.NoError(t, err)
-	return resp
-}
-
-func queryTesterReceivedErr(t *testing.T, chain *wasmibctesting.TestChain, tester string) error {
-	resp := TesterResponse{}
-	testerReceivedQuery := TesterReceivedQuery{
-		GetReceivedCallback: EmptyData{},
-	}
-	err := chain.SmartQuery(tester, testerReceivedQuery, &resp)
-	return err
-}
-
-func (suite *CbTestSuite) TestSuccessfulTransfer() {
+func (suite *CallbackTestSuite) TestSuccessfulTransfer() {
 	memo := callbackMemo(nftSentCb(), "", nftReceivedCb(), "")
+	// A -> B token_id 2
 	sendIcsFromChainAToB(suite, suite.cw721A.String(), "2", memo, true)
 
 	// Query the owner of NFT on cw721
-	chainAOwner := queryGetOwnerOf(suite.T(), suite.chainA, suite.cw721A.String(), "2")
+	chainAOwner := test_suite.QueryGetOwnerOf(suite.T(), suite.chainA, suite.cw721A.String(), "2")
 	require.Equal(suite.T(), chainAOwner, suite.bridgeA.String())
-	chainBOwner := queryGetOwnerOf(suite.T(), suite.chainB, suite.cw721B.String(), "2")
+	chainBOwner := test_suite.QueryGetOwnerOf(suite.T(), suite.chainB, suite.cw721B.String(), "2")
 	require.Equal(suite.T(), chainBOwner, suite.testerB.String())
 
 	// We query the data we have on the tester contract
 	// This ensures that the callbacks are called after all the messages was completed
 	// and the transfer was successful
-	testerDataOwnerA := queryTesterSent(suite.T(), suite.chainA, suite.testerA.String())
+	testerDataOwnerA := test_suite.QueryTesterSent(suite.T(), suite.chainA, suite.testerA.String())
```

Comment:
No clear how below 3 assertions work. In case ack and receive addr is empty in callback, then ack callback is called on sender on chain A, and receive callback is called on recipient. So some kind of result data is check and verified on sender and recipient (contract?)?